go/vt/vtgate/evalengine/translate.go

```diff
@@ -402,20 +402,32 @@ func translateUnaryExpr(unary *sqlparser.UnaryExpr, lookup TranslationLookup) (E
 	}
 }
 
-func translateConvertCharset(charset string, lookup TranslationLookup) (collations.ID, error) {
+func binaryCollationForCollation(collation collations.ID) collations.ID {
```

Comment:
Dunno if there's a more efficient way to do this, or if we should care to be more efficient here for this fairly edge case behavior?